### ISSUE TITLE
The ios_config module does not delete the username from the router

### ISSUE THREAD

**Issue by ubajze** [ansible/ansible-modules-core#3591]
##### ISSUE TYPE

- Bug Report
##### COMPONENT NAME

networking/ios_config
##### ANSIBLE VERSION

```
$ ansible --version
ansible 2.0.1.0
```
##### CONFIGURATION

No configuration.
##### OS / ENVIRONMENT

Working on Ubuntu, release trusty. The issue is not platform dependent.
##### SUMMARY

When using command "no username" with ios_config module, the task timeouts. 
##### STEPS TO REPRODUCE

I am using the following task configuration in the playbook. 

```
    - name: Delete users
      ios_config:
        host: "{{ ansible_ssh_host }}"
        username: "{{ username }}"
        password: "{{ password }}"
        lines:
          - no username admin
```

The result of the task is:

```
TASK [Delete users] ************************************************************
fatal: [hub]: FAILED! => {"changed": false, "commands": ["configure terminal", "no username admin"], "failed": true, "msg": "timeout trying to send command"}
```

The reason for this is that Cisco ask for confirmation when deleting the username from the configuration.

```
R2(config)#no username admin
This operation will remove all username related configurations with same name.Do you want to continue? [confirm]
R2(config)#
```

##### EXPECTED RESULTS

It is expected that command should be deleted from the configuration.
##### ACTUAL RESULTS

```
fatal: [wan1]: FAILED! => {"changed": false, "commands": ["configure terminal", "no username admin"], "failed": true, "invocation": {"module_args": {"after": null, "auth_pass": null, "authorize": false, "before": null, "config": null, "force": false, "host": "192.168.35.152", "lines": ["no username admin"], "match": "line", "parents": null, "password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER", "port": 22, "provider": null, "replace": "line", "username": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER"}, "module_name": "ios_config"}, "msg": "timeout trying to send command"}
```

**Comment by nitzmahone** [comment]
cc @Qalthos @privateip

**Comment by dabruzzo** [comment]
Same problem for ios_command module.

**Comment by ansibot** [comment]
@privateip, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/jctanner/ansibullbot/blob/ISSUE_TRIAGE_CLASS/ISSUE_HELP.md)

**Comment by gundalow** [comment]
@privateip and I reviewed this today. If we get time for 2.2 we would like to do this, however it is a low priority.

**Comment by ansibot** [comment]
@privateip, @gundalow, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@privateip, @gundalow, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@privateip, @gundalow, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@privateip, @gundalow, @qalthos, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by gundalow** [comment]
We will fix this in 2.3, this will not be backported to 2.2.

**Comment by dabruzzo** [comment]
Look forward to update to 2.3 release.

Thanks

--
Dario

**Comment by ansibot** [comment]
This repository has been locked. All new issues and pull requests should be filed in https://github.com/ansible/ansible

Please read through the <a href="http://docs.ansible.com/ansible/dev_guide/repomerge.html#move-issues-and-prs-to-new-repo">repomerge</a> page in the dev guide. The guide contains links to tools which automatically move your issue or pull request to the ansible/ansible repo.

**Comment by ansibot** [comment]
This issue was migrated to https://github.com/ansible/ansible/issues/29807
